### PR TITLE
Dont decode S3 Bucket policies

### PR DESCRIPTION
**Pull request by patrobinson**
This fixes an issue that was introduced in #13. ~I can't really figure out why we would want to decode the Policy, it should never be encoded in the first place as far as I can tell.~
S3 bucket policies are not URL encoded, while other policies are.
If the bucket policy contains invalid encoding, such as `%` followed by non-hexadecimal characters, it creates an error such as:
```
Error decoding policy invalid URL escape "%zz"
```

### CODE REVIEW

File: iamy/policy_test.go [hunk 1]
```diff
@@ -126,3 +126,10 @@ Actual:  %#v`, nt.description, nt.input, nt.expected, result)
 		}
 	}
 }
+
+func TestNewPolicyDocumentFromJson(t *testing.T) {
+	_, err := NewPolicyDocumentFromJson("{\"Version\":\"2012-10-17\",\"Id\":\"AllowPublicRead\",\"Statement\":[{\"Sid\":\"PublicReadBucketObjects\",\"Effect\":\"Allow\",\"Principal\":\"*\",\"Action\":\"s3:GetObject\",\"Resource\":\"arn:aws:s3:::example.com/*\",\"Condition\":{\"StringEquals\":{\"aws:Referer\":\"%zz\"}}}]}")
```
Comment (mtibben) [inline iamy/policy_test.go]: Use backticks for the string here for readability

File: iamy/aws.go [hunk 2]
```diff
@@ -120,8 +120,11 @@ func (a *AwsFetcher) fetchIamData() error {
 			}),
 		},
 		func(resp *iam.GetAccountAuthorizationDetailsOutput, lastPage bool) bool {
+			// There's a bug here. This doesn't set the variable outside this function scope
+			// So IAMY just swallows any errors returned from populateIamData()
```
Comment (mtibben) [inline iamy/aws.go]: hmm I'm not sure I understand... can you fix or produce a testcase?

File: iamy/policy.go [hunk 3]
```diff
@@ -8,16 +8,28 @@ import (
 	"sort"
 )
 
+func NewPolicyDocumentFromJson(jsonString string) (*PolicyDocument, error) {
+	var doc PolicyDocument
+	if err := json.Unmarshal([]byte(jsonString), &doc); err != nil {
+		log.Printf("Error unmarshalling JSON %s %s", err, jsonString)
+		return nil, err
+	}
+	//log.Printf("Doc %s\nJSON: %s", doc, jsonString)
```
Comment (mtibben) [inline iamy/policy.go]: Remove commented out code

File: iamy/policy.go [hunk 4]
```diff
@@ -8,16 +8,28 @@ import (
...
 		return nil, err
 	}
-
 	var doc PolicyDocument
-	if err = json.Unmarshal([]byte(jsonString), &doc); err != nil {
+	if err := json.Unmarshal([]byte(jsonString), &doc); err != nil {
+		log.Printf("Error unmarshalling JSON %s %s", err, jsonString)
 		return nil, err
 	}
+	//log.Printf("Doc %s\nJSON: %s", doc, jsonString)
```
Comment (mtibben) [inline iamy/policy.go]: Remove commented out code

File: iamy/aws.go (hunk 2 above)
Comment (patrobinson) [reply iamy/aws.go]: I tried to fix it, but I'm not entirely sure. I'll pull this out into a seperate WIP PR and see if I can write a failing test.

**Comment by patrobinson** [comment]
Thanks for the review @mtibben I've addressed your feedback and given this a whirl in one of our accounts. Let me know your thoughts

**Pull request merged**